From 3f06c1cf80b2dd2abf7354e13d64241fbc43fdaa Mon Sep 17 00:00:00 2001
From: Han Xu <han.xu@nxp.com>
Date: Thu, 7 Apr 2016 10:22:57 -0500
Subject: [PATCH 0996/5242] MLK-12622: mtd: gpmi: fix the issue in legacy bch
 support

commit  5af20397ec68fd67dc69621c7a7a704d547964e2 from
https://source.codeaurora.org/external/imx/linux-imx.git

missed the brackets for bch legacy support, which leads the large oob
nand bch setting to wrong path.

Signed-off-by: Han Xu <han.xu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c b/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c
index 5e02c1b..698c63f 100644
--- a/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c
+++ b/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c
@@ -613,9 +613,10 @@ int common_nfc_set_geometry(struct gpmi_nand_data *this)
 	}
 
 	if ((!(chip->ecc_strength_ds > 0 && chip->ecc_step_ds > 0) &&
-			(mtd->oobsize < 1024)) || this->legacy_bch_geometry)
+			(mtd->oobsize < 1024)) || this->legacy_bch_geometry) {
 		dev_warn(this->dev, "use legacy bch geometry\n");
 		return legacy_set_geometry(this);
+	}
 
 	if (mtd->oobsize > 1024 || chip->ecc_step_ds < mtd->oobsize)
 		return set_geometry_for_large_oob(this);
-- 
1.7.9.5

